savant/deepstream/nvinfer/element_config.py

```diff
@@ -348,36 +348,48 @@ def process(self, msg, kwargs):
 
     # model type-specific parameters
     if issubclass(model_type, ObjectModel):
-        if not model_config.output.objects:
-            # try to load model object labels from file
-            label_file = model_config.get(
-                'label_file',
-                nvinfer_config['property'].get('labelfile-path')
-                if nvinfer_config
-                else None,
-            )
+        # model_config.output.objects is mandatory for object models
+        # but it may be autogenerated based on labelfile or num_detected_classes
+
+        label_file = model_config.get(
+            'label_file',
+            nvinfer_config['property'].get('labelfile-path')
+            if nvinfer_config
+            else None,
+        )
+        if model_config.output.objects:
+            # highest priority is using manually defined model_config.output.objects
             if label_file:
-                label_file_path = model_path / label_file
-                if label_file_path.is_file():
-                    with open(label_file_path.resolve(), encoding='utf8') as file_obj:
-                        model_config.output.objects = [
-                            NvInferObjectModelOutputObject(
-                                class_id=class_id, label=label
-                            )
-                            for class_id, label in enumerate(
-                                file_obj.read().splitlines()
-                            )
-                        ]
-                        model_config.output.num_detected_classes = len(
-                            model_config.output.objects
-                        )
-                    logger.info(
-                        'Model object labels have been loaded from "%s".',
-                        label_file_path,
-                    )
+                logger.warning(
+                    'Model output objects labels are defined manually '
+                    'and will be used instead of labels from "%s".',
+                    label_file,
+                )
 
-        # generate labels (enumerate)
-        if not model_config.output.objects and model_config.output.num_detected_classes:
+        elif label_file:
+            # try to load model object labels from file
+            label_file_path = model_path / label_file
+            if label_file_path.is_file():
+                with open(label_file_path.resolve(), encoding='utf8') as file_obj:
+                    model_config.output.objects = [
+                        NvInferObjectModelOutputObject(class_id=class_id, label=label)
+                        for class_id, label in enumerate(file_obj.read().splitlines())
+                    ]
+                if model_config.output.num_detected_classes:
```

Comment:
I think no need for warning if model_config.output.num_detected_classes == len(model_config.output.objects)

Reply:
This warning is about the config node not being used at all, regardless of the actual value.

So I don't think additional check for the case when manually set value coincides with the number of labels in the label file is needed.